### PR TITLE
Flake8 all the things

### PR DESCRIPTION
**Pull request by kurtschelfthout**
Removed pylint (which we were not using), as after testing it felt pretty heavy/slow and had overlap with mypy.
Flake8 does the two things that I wanted from it pretty well:
- check for unused imports (despite your best efforts a few were remaining :) )
- check for line length of comments/strings

For the latter it's a bit annoying that half of the `noqa` ignores are actuallly due to mypy ignores, but I guess that's just extra incentive to fix those at some point. In any case this didn't feel too onerous to fix.

This hopefully saves you from having to check that stuff manually in the future!

### CODE REVIEW

File: src/meadowdb/connection.py [hunk 1]
```diff
@@ -94,8 +96,8 @@ def set_default_userspace_spec(
 @dataclasses.dataclass(frozen=True)
 class MeadowdbEffects:
     """
-    This should be part of meadowflow.effects, but we want meadowflow to depend on
-    meadowdb, not the other way around.
+    This should be part of meadowflow.effects, but we want meadowflow to depend
```
Comment (hrichardlee) [inline src/meadowdb/connection.py]: I think you might have your comment reformatter set to 80 characters? I think this comment was fine before the change? (Under 88 characters)

File: src/meadowdb/connection.py [hunk 2]
```diff
@@ -202,16 +206,15 @@ def delete_where_equal(
...
-        delete_where_equal_df will be "deleted". E.g. if delete_where_equal_df is
+        delete_where_equal_df should also be a pandas dataframe that has a
+        subset of the columns in table_name. Any existing rows matching the
+        values in delete_where_equal_df will be "deleted". E.g. if
+        delete_where_equal_df is
 
-        a | b
-        1 | 2
-        3 | 4
+        a | b 1 | 2 3 | 4
```
Comment (hrichardlee) [inline src/meadowdb/connection.py]: Need to preserve formatting!

**Review by hrichardlee — COMMENTED** [review]
Makes sense, thank you!

File: src/meadowdb/connection.py (hunk 1 above)
Comment (kurtschelfthout) [reply src/meadowdb/connection.py]: Yeah, took me a while initially to configure everything to 88, should be good now.

**Review by hrichardlee — APPROVED** [review]

**Pull request merged**